tests/functions/test_map_to_s3.py

```diff
@@ -0,0 +1,134 @@
+from cdn_lambda.functions.map_to_s3 import lambda_handler
+import os
+import mock
+import json
+
+
+@mock.patch("boto3.client")
+@mock.patch("cdn_lambda.functions.map_to_s3.datetime")
+def test_map_to_s3(mocked_datetime, mocked_boto3_client):
+    mocked_datetime.utcnow.return_value = "2020-02-12 20:55:06.372878"
+    mocked_boto3_client().query.return_value = {
```

Comment:
Have you seen moto library? https://github.com/spulec/moto

I think we should give that a try for the tests.  I'm really not a fan of the mocking approach here, because tests using these mocks are very weak.  mocked_boto3_client().query.assert_called_with(absolute_garbage) is going to pass if both the test and the code under test use the same garbage.

As I understand it, with moto we could use the regular boto3 API to insert records into a DynamoDB, then the queries in our lambda code should work normally instead of going into a mock which doesn't do anything.  We can then get by just with testing the outcome of the lambda.

That being said, I don't know how complete moto's DynamoDB backend is, so someone would have to give it a try and see if it's workable.

Reply:
While it looks promising, moto doesn't "just work" apparently. I can't even successfully put an item into a DB table. Also, looking through the issues on Github, it seems there are a lot of behavioral inconsistencies.